### PR TITLE
Add conversion methods to Request

### PR DESCRIPTION
**Pull request by Gallaecio**
https://github.com/scrapinghub/web-poet/pull/206#issuecomment-2225683651

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -205,18 +205,28 @@ Using request data
...
 its output request data to build an actual request with any HTTP client
-software, like the requests_ library (see an example :ref:`above
-<requests-example>`) or the :doc:`Scrapy <scrapy:index>` web scraping
-framework:
+software.
+
+:class:`~form2request.Request` also provides conversion methods for common use
+cases:
+
+-   :class:`~form2request.Request.to_poet`, for :doc:`web-poet 0.2.0+
```
Comment (kmike) [inline docs/usage.rst]: I think we should reverse the order: Scrapy, then requests, then web-poet.

File: form2request/_base.py [hunk 2]
```diff
@@ -178,6 +178,57 @@ class Request:
     headers: list[tuple[str, str]]
     body: bytes
 
+    def to_poet(self, **kwargs):
```
Comment (kmike) [inline form2request/_base.py:181]: That's a tricky one, but is it possible to type annotate these methods, without making the dependencies non-optional? I'm fine with merging without it.

**Review by kmike — APPROVED** [review]

File: form2request/_base.py (hunk 2 above)
Comment (Gallaecio) [reply form2request/_base.py:181]: I have found some options, but I’m not sure about them:
- [Importing in `TYPE_CHECKING`](https://stackoverflow.com/a/61394121/939364) (1st part of the response), which would make extras required during type checking.
- [Only defining the methods when the extras are importable](https://stackoverflow.com/a/61394121/939364) (2nd part of the response), which at the very least could be confusing for users.
- [Using a TypeVar when the extra is unavailable](https://stackoverflow.com/a/72440529/939364) looks OK to me, not sure if there is any significant drawback not mentioned yet in StackOverflow.

Comment (kmike) [reply form2request/_base.py:181]: Regarding the first approach, with TYPE_CHECKING. 
Would it require a `requests` user to have Scrapy installed if they want to check their code (which uses to_requests), taking form2requests definitions in account?

Comment (Gallaecio) [reply form2request/_base.py:181]: I think so, yes. Unintuitively, I listed options from my least favorite to my least “not favorite”.

Comment (Gallaecio) [reply form2request/_base.py:181]: Option 3 does not seem to work.

```
form2request/_base.py:19: error: Invalid assignment target  [misc]
form2request/_base.py:205: error: A function returning TypeVar should receive at least one argument containing the same TypeVar  [type-var]
form2request/_base.py:219: error: Incompatible return value type (got "PreparedRequest", expected "PreparedRequest")  [return-value]
```

Comment (kmike) [reply form2request/_base.py:181]: Let's skip it then :)

**Pull request merged**